core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSource.scala

```diff
@@ -438,6 +541,9 @@ case class DeltaSource(
     initialState.iterator()
   }
 
+  /**
+   * Narrow-waist for generating snapshot from Delta Log within Delta Source
```

Comment:
what does narrow-waist mean in this context?

Reply:
This is just to note, because I believe this function is the ultimate function for generating snapshots across all places of DeltaSource, so any new usage should adopt this util.